koku/masu/external/downloader/azure/azure_service.py

```diff
@@ -64,24 +64,31 @@ def __init__(
             raise AzureServiceError("Azure Service credentials are not configured.")
 
     def _get_latest_blob(
-        self, report_path: str, blobs: list[BlobProperties], extension: str
+        self, report_path: str, blobs: list[BlobProperties], extension: t.Optional[str] = None
```

Comment:
Should we even be passing the extensions around these methods? Michael already mentioned that it was only used in the unit tests anyway. That said I just double checked and we are using here:

`return self._get_latest_blob_for_path(report_path, container_name, AzureBlobExtension.manifest.value)`

BUT I'm not even convinced that does what we think it should! That extension is completely incorrect and we never find a manifest.

In [1]: from masu.util.azure.common import AzureBlobExtension

In [2]: AzureBlobExtension.manifest.value
Out[2]: '_manifest.json'

Now that I'm thinking about it we shouldnt even be passing extensions/compression types around at all if we actually downloaded the manifest correctly then it could just be read from that.

So it turns out that Azure export types suck! -->
- Partitioned exports have `_manifest.json`
- Gzip compressed exports have `manifest.json`
- Standard non-compressed exports don't have a manifest at all.

Reply:
I changed this extension to `AzureBlobExtension.json.value`. In my tests it has downloaded reports on gz, csv, with and without the manifest file.

We have some duplicated logic here. Sometimes (if the manifest file does not exist) it downloads a file twice (in `_get_manifest` and in `download_file` methods). It is needed because we have to know the report name and other details before download it (it is a bit confusing specially when there is no manifest file). FMPOV I think it is working but we can rethink this logic if we want.